external_parser/parse_example_external.cc

```diff
@@ -29,9 +34,11 @@ parser::get_external_parser(vw *all, const input_options &parsed_options) {
       }
 
       std::string outfile_name = infile_name + ".dsjson";
```

Comment:
why not construct unique pointers via `VW::make_unique` and have the `binary_parser` constructor take a `unique_ptr` which it will then move into the `_example_joiner` member?

Reply:
do you mean to let constructor to have unique_ptr&& as input argument?